src/local.rs

```diff
@@ -447,4 +506,31 @@ mod tests {
             panic!("unexpected error type: {:?}", err);
         }
     }
+
+    #[tokio::test]
+    async fn root() {
+        let integration = LocalFileSystem::new();
+
+        let canonical = std::path::Path::new("Cargo.toml").canonicalize().unwrap();
+        let url = Url::from_directory_path(canonical).unwrap();
+        let path = Path::from_raw(url.path());
+
+        integration.head(&path).await.unwrap();
```

Comment:
the test is that this doesn't error (namely that path --> url --> path is valid?)

Is it also worth checking that `canonical == path`?